calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -216,6 +221,217 @@ def test_validate_arguments(self, case, sys_exit_called):
+        ]
+        binds = {
+            "/proc":
+                {
+                    "bind": "/proc_host",
+                    "ro": False
+                },
+            log_dir:
+                {
+                    "bind": "/var/log/calico",
+                    "ro": False
+                },
+            "/run/docker/plugins":
+                {
+                    "bind": "/usr/share/docker/plugins",
+                    "ro": False
+                }
+        }
+
+        # Assert
+        m_os.path.exists.assert_called_once_with('./log_dir')
+        m_check_system.assert_called_once_with(fix=False, quit_if_error=False)
+        m_get_host_ips.assert_called_once_with(4)
+        m_warn_if_unknown_ip.assert_called_once_with(ip_2, ip6)
+        m_warn_if_hostname_conflict.assert_called_once_with(ip_2)
+        m_install_kube.assert_called_once_with(node.KUBERNETES_PLUGIN_DIR)
+        m_client.get_ip_pools.assert_has_calls([call(4), call(6)])
+        m_client.ensure_global_config.assert_called_once_with()
+        m_client.create_host.assert_called_once_with(
+            node.hostname, ip_2, ip6, as_num
+        )
+        m_docker_client.remove_container.assert_called_once_with(
+            'calico-node', force=True
+        )
+        m_os.getenv.assert_called_once_with(
+            ETCD_AUTHORITY_ENV, ETCD_AUTHORITY_DEFAULT
+        )
+        m_sys_exit.assert_called_once_with(1)
+        m_docker.utils.create_host_config.assert_called_once_with(
+            privileged=True,
+            restart_policy={"Name":"Always"},
+            network_mode="host",
+            binds=binds
+        )
+        m_find_or_pull_node_image.assert_called_once_with(
+            'node_image', m_docker_client
+        )
+        m_docker_client.create_container.assert_called_once_with(
+            node_image,
+            name='calico-node',
+            detach=True,
+            environment=environment,
+            host_config='host_config',
+            volumes=['/proc_host',
+                     '/var/log/calico',
+                     '/usr/share/docker/plugins']
+        )
+        m_docker_client.start.assert_called_once_with(container)
+        m_attach_and_stream.assert_called_once_with(container)
+
+
+    @patch('sys.exit', autospec=True)
+    @patch('calico_ctl.node.os', autospec=True)
+    @patch('calico_ctl.node.check_system', autospec=True)
+    @patch('calico_ctl.node.get_host_ips', autospec=True)
+    @patch('calico_ctl.node.warn_if_unknown_ip', autospec=True)
+    @patch('calico_ctl.node.warn_if_hostname_conflict', autospec=True)
+    @patch('calico_ctl.node.install_kubernetes', autospec=True)
+    def test_node_start_oserror(self, m_install_kube, m_warn_if_hostname_conflict,
+                                m_warn_if_unknown_ip, m_get_host_ips,
+                                m_check_system, m_os, m_sys_exit):
+        """
+        Test node_start function in calicoctl node command
```

Comment:
What failure case is this covering?

It isn't apparent from the docstring, nor the title, except that it somehow involves OSErrors ;)